korangar/src/graphics/passes/directional_shadow/shader/entity_bindless.wgsl

```diff
@@ -47,26 +51,35 @@ fn vs_main(
 ) -> VertexOutput {
     let instance = instance_data[instance_index];
     let vertex = vertex_data(vertex_index);
+    let new_vertex = instance.affine * vec4<f32>(vertex.position, 1.0);
+    let world_position = instance.world * new_vertex;
 
     var output: VertexOutput;
-    output.position = pass_uniforms.view_projection * instance.world * vec4<f32>(vertex.position, 1.0);
+    output.position = pass_uniforms.view_projection * instance.world * new_vertex;
     output.texture_coordinates = instance.texture_position + vertex.texture_coordinates * instance.texture_size;
 
     if (instance.mirror != 0u) {
         output.texture_coordinates.x = 1.0 - output.texture_coordinates.x;
     }
 
-    output.depth_offset = vertex.depth_multiplier;
-    output.curvature = vertex.curvature_multiplier;
+    output.depth_offset = new_vertex.y/2.0 + instance.extra_depth_offset;
```

Comment:
This looks odd. Why are you now using the vertex x/y coordinates here?

Reply:
It is possible to calculate the depth and curvature using the x/y-coordinate. For depth_offset is y/2.0 and for curvature is x. In the original function of vertex it could be changed there also.

Reply:
Then you need to explain WHY this is possible in the code. Is it because it correlates to the X/Y value? I think the correct place for this is the vertex_data. Then you can also document it in the truth table.

Reply:
Oh wait ... I think there is no need for chaneg the vertex_data() function at all and just us the old values (depth_multiplier and curvature_multiplier), since they have exactly the same value as you re-calculate here!

Reply:
You need to recalculate the depth and curvature unfortunately after the frame part transform. Another way is using a matrix multiplication with odd size to gather the result of depth and curvature, but it is also a problem because of the padding format in the shader side.

Reply:
Then I write exactly what you just told me! The comment would be better this way:
```
// Because we have to transform the vertex of the frame part, we can't use the depth and curvature
// directly and are using the fast, that y/depth and x/curvature correlate to each other.
```

Reply:
Added the comment and also explained the offset.

Reply:
Great! LGTM!